qutebrowser/browser/hints.py

```diff
@@ -438,14 +464,24 @@ def _click(self, elem, context):
             target_mapping[Target.tab] = usertypes.ClickTarget.tab_bg
         else:
             target_mapping[Target.tab] = usertypes.ClickTarget.tab
+
         # FIXME Instead of clicking the center, we could have nicer heuristics.
         # e.g. parse (-webkit-)border-radius correctly and click text fields at
         # the bottom right, and everything else on the top left or so.
         # https://github.com/The-Compiler/qutebrowser/issues/70
-        pos = elem.rect_on_view().center()
+        rect = self._get_first_rectangle(elem)
+        pos = rect.center()
+
+        # fix coordinates according to zoom level
+        zoom = elem.webFrame().zoomFactor()
+        if not config.get('ui', 'zoom-text-only'):
+            pos.setX(pos.x() * zoom)
+            pos.setY(pos.y() * zoom)
```

Comment:
I think this accounts for the zoom twice if `_get_first_rectangle` falls back on `rect_on_view`?